src/app/txRunner.js

```diff
@@ -65,25 +65,26 @@ TxRunner.prototype.execute = function () {
         })
       })
     } else {
-      self.web3.eth.estimateGas(tx, function (err, resp) {
+      self.web3.eth.estimateGas(tx, function (err, gasEstimation) {
         if (err) {
-          return callback(err, resp)
+          return callback(err, gasEstimation)
         }
+        self.web3.eth.getBlock('latest', function (err, block) {
+          if (err) {
+            return callback(err)
+          } else {
+            var blockGasLimit = Math.floor(block.gasLimit - block.gasLimit / 1024)
+            tx.gas = blockGasLimit < gasEstimation ? blockGasLimit : gasEstimation
 
-        if (resp > gasLimit) {
-          return callback('Gas required exceeds limit: ' + resp)
```

Comment:
I think this is useful, because the user can set a hard limit (`gasLimit`) in the UI.